internal/cli/federatedauthentication/federationsettings/connectedorgsconfigs/connect.go

```diff
@@ -0,0 +1,137 @@
+// limitations under the License.
+
+package connectedorgsconfigs
+
+import (
+	"context"
+
+	"github.com/mongodb/mongodb-atlas-cli/atlascli/internal/cli"
+	"github.com/mongodb/mongodb-atlas-cli/atlascli/internal/config"
+	"github.com/mongodb/mongodb-atlas-cli/atlascli/internal/flag"
+	"github.com/mongodb/mongodb-atlas-cli/atlascli/internal/store"
+	"github.com/mongodb/mongodb-atlas-cli/atlascli/internal/usage"
+	"github.com/spf13/cobra"
+	atlasv2 "go.mongodb.org/atlas-sdk/v20231115012/admin"
+)
+
+type ConnectOpts struct {
+	cli.GlobalOpts
+	cli.OutputOpts
+	cli.InputOpts
+	*DescribeOrgConfigsOpts
+	identityProviderID   string
+	protocol             string
+	federationSettingsID string
+	store                store.ConnectedOrgConfigsUpdater
+}
+
+const (
+	oidc = "OIDC"
+	saml = "SAML"
+)
+
+const updateTemplate = `Connected Org Config with {{range $index, $element := .DataAccessIdentityProviderIds}}{{if $index}}, {{end}}{{$element}}{{end}}.`
+
+func (opts *ConnectOpts) InitStore(ctx context.Context) func() error {
+	return func() error {
+		if opts.store != nil && opts.describeStore != nil {
+			return nil
+		}
+
+		var err error
+		opts.store, err = store.New(store.AuthenticatedPreset(config.Default()), store.WithContext(ctx))
+		if err != nil {
+			return err
+		}
+
+		return opts.InitDescribeStore(ctx)()
+	}
+}
+
+func (opts *ConnectOpts) Run() error {
+	var orgConfig *atlasv2.ConnectedOrgConfig
+	var err error
+	if orgConfig, err = opts.GetConnectedOrgConfig(opts.federationSettingsID, opts.ConfigOrgID()); err != nil {
+		return err
+	}
+
+	if len(orgConfig.GetRoleMappings()) == 0 {
+		orgConfig.RoleMappings = nil
+	}
+
+	params := &atlasv2.UpdateConnectedOrgConfigApiParams{
+		FederationSettingsId: opts.federationSettingsID,
+		OrgId:                opts.ConfigOrgID(),
+		ConnectedOrgConfig:   orgConfig,
+	}
+
+	if opts.protocol == oidc {
+		if orgConfig.DataAccessIdentityProviderIds == nil {
+			orgConfig.DataAccessIdentityProviderIds = &[]string{}
+		}
+
+		newList := append(*orgConfig.DataAccessIdentityProviderIds, opts.identityProviderID)
+		params.ConnectedOrgConfig.DataAccessIdentityProviderIds = &newList
+	} else if opts.protocol == saml {
+		params.ConnectedOrgConfig.IdentityProviderId = &opts.identityProviderID
+	}
+
+	r, err := opts.store.UpdateConnectedOrgConfig(params)
+	if err != nil {
+		return err
+	}
+
+	return opts.Print(r)
+}
+
+// atlas federatedAuthentication connectedOrgs connect --identityProviderId identityProviderId --federationSettingsId federationSettingsId [-o/--output output].
+func ConnectBuilder() *cobra.Command {
+	opts := &ConnectOpts{
+		DescribeOrgConfigsOpts: &DescribeOrgConfigsOpts{},
+	}
+	cmd := &cobra.Command{
+		Use:   "connect",
+		Short: "Connect an Identity Provider to an Organization.",
+		Args:  cobra.NoArgs,
+		Example: `  # Connect the current profile Org with Identity Provider with ID 5d1113b25a115342acc2d1aa and  federationSettingsId 5d1113b25a115342acc2d1aa 
```

Comment:
extra space and I think `org` should be all lowercase. Changing one instance of `with` as well to limit repetition
```suggestion
		Example: `  # Connect the current profile org to identity provider with ID 5d1113b25a115342acc2d1aa and federationSettingsId 5d1113b25a115342acc2d1aa 
```